content/en/community/hackathons/sessions/behind-scenes/content/Work-Items_06.-Tutorial---Reminder:-Adding-Filesystems-to-an-Application-2023.md

```diff
@@ -0,0 +1,136 @@
+For this tutorial, the aim is to create a simple QEMU / KVM application that reads from a file and displays the contents to standard output.
+A local directory is to be mounted as the root directory (`/`) inside the QEMU / KVM virtual machine.
+#### Setup
+
+The basic setup is in the `work/03-adding-filesystems/` folder in the session directory. 
+
+**If, at any point of this tutorial, something doesn't work, or you want a quick check, see the reference solution in `sol/03-adding-filesystems/` folder in the session directory.**
+
+1. Enter the `work/03-adding-filesystems/:
+
+    ```console
+    $ cd work/03-adding-filesystems/
+
+    $ ls -F
+    guest_fs/  kraft.yaml  launch.sh*  main.c  Makefile  Makefile.uk  qemu-guest*
+    ```
+
+    * The `guest_fs/` local directory is to be mounted as the root directory (`/`) inside the QEMU / KVM virtual machine.
+    It contains the `grass` file.
+    * The program (`main.c`) reads the contents of the `/grass` file and prints it to standard output.
+    * `Makefile.uk` lists the `main.c` file as the application source file to be compiled and linked with Unikraft.
+    * `Makefile` is used by the manual configuration and build system.
+    * `launch.sh` is a wrapper script around `qemu-system-x86_64` used to manually run the application.
+We'll use it as well to run the application.
+
+
+1. Clone the Unikraft repository:
+
+    ```console
+    $ mkdir .unikraft
+    $ cd .unikraft/
+    $ mkdir libs/
+    $ cd ../
+    ```
+
+1. Add the `lib-musl` repository:
+
+    For filesystem functionalities (opening, reading, writing files) we require a more powerful libc.
+    [Musl](https://github.com/unikraft/lib-musl) is already ported in Unikraft and will do nicely:
+
+    ```console
+    $ cd .unikraft/libs
+    $ git clone https://github.com/unikraft/lib-musl musl
+    ```
+
+    We also need to update the `LIBS` line in the `Makefile`:
+
+    ```console
+    LIBS := $(UK_LIBS)/musl
+    ```
+
+### Configuring the application
+
+Now we need to enable **9pfs** and **musl** in Unikraft.
+To do this, we run:
+
+```console
+$ make menuconfig
+```
+
+We need to select the following options, from the `Library Configuration` menu:
+
+* `libmusl`
```

Comment:
In menuconfig, it shows 'musl: A C Standard library',. Maybe we can change from 'libmusl' to this text.